File: recSys/cpp/src/main/models/baseline/PopularityModel.cpp
#include "PopularityModel.h"

double PopularityModel::prediction(RecDat* rec_dat){
  int prediction = items_.get(rec_dat->item);
  if (prediction == 0) return 0;
  else return log(prediction+1)/log(maxitem_+1); //maxitem_>0 -> log(maxitem_+1)>0
}

File: python/alpenglow/offline/models/PopularityModel.py
import alpenglow.Getter as rs
import alpenglow.offline


class PopularityModel(alpenglow.offline.OfflineModel):
    """PopularityModel()

    Recommends the most popular item from the set of items.
    """
    def _fit(self, recommender_data, users, items, matrix):
        model = rs.PopularityModel()

        updater = rs.PopularityModelUpdater()
        updater.set_model(model)

        learner = rs.OfflineIteratingOnlineLearnerWrapper(
            seed=0,
            number_of_iterations=0,
            shuffle=False,
        )
        learner.add_updater(updater)

        return (model, learner)
